src/audio/copier/copier_hifi.c

```diff
@@ -72,4 +74,67 @@ int apply_attenuation(struct comp_dev *dev, struct copier_data *cd,
 		return -EINVAL;
 	}
 }
+
+void copier_gain_set_basic_params(struct comp_dev *dev, struct dai_data *dd)
+{
+	struct copier_gain_params *gain_params = dd->gain_data;
+	struct ipc4_copier_module_cfg *copier_cfg = dd->dai_spec_config;
+
+	/* Set default gain coefficients */
+	for (int i = 0; i < ARRAY_SIZE(gain_params->gain_coeffs); ++i)
+		gain_params->gain_coeffs[i] = AE_MOVF16X4_FROMINT64(UNITY_GAIN_4X_Q10);
+
+	gain_params->step_f16 = AE_ZERO16();
+	gain_params->init_gain = AE_ZERO16();
+
+	/* Set audio format */
+	gain_params->container = copier_cfg->base.audio_fmt.depth;
+	gain_params->chanels_count = copier_cfg->base.audio_fmt.channels_count;
+}
+
+int copier_gain_set_fade_params(struct comp_dev *dev, struct dai_data *dd,
+				uint32_t fade_period, uint32_t frames)
+{
+	struct copier_gain_params *gain_params = dd->gain_data;
+	struct ipc4_copier_module_cfg *copier_cfg = dd->dai_spec_config;
+	uint16_t init_gain[4];
+	ae_f16 step_f16;
+
+	/* For backward compatibility add a case with default fade transition.
+	 * Backward compatibility is referring to clock_on_delay in DMIC blob.
+	 */
+	if (fade_period == 0) {
+		/* Default fade transition delay for speech */
```

Comment:
there's nothing in this PR that allows a configuration of speech v. some other type of content.